front/lib/resources/file_resource.ts

```diff
@@ -0,0 +1,228 @@
+  static model: ModelStatic<FileModel> = FileModel;
+
+  constructor(model: ModelStatic<FileModel>, blob: Attributes<FileModel>) {
+    super(FileModel, blob);
+  }
+
+  static async makeNew(
+    blob: Omit<CreationAttributes<FileModel>, "status" | "sId">
+  ) {
+    const fileId = makeDustFileId();
+
+    const key = await FileResource.model.create({
+      ...blob,
+      status: "created",
+      sId: fileId,
+    });
+
+    return new this(FileResource.model, key.get());
+  }
+
+  static async fetchById(
+    auth: Authenticator,
+    id: FileId
+  ): Promise<FileResource | null> {
+    // TODO(2024-07-01 flav) Remove once we introduce AuthenticatorWithWorkspace.
+    const owner = auth.workspace();
+    if (!owner) {
+      throw new Error("Unexpected unauthenticated call to `getUploadUrl`");
+    }
+
+    const blob = await this.model.findOne({
+      where: {
+        workspaceId: owner.id,
+        sId: id,
+      },
+    });
+    if (!blob) {
+      return null;
+    }
+
+    // Use `.get` to extract model attributes, omitting Sequelize instance metadata.
+    return new this(this.model, blob.get());
+  }
+
+  static async deleteAllForWorkspace(
+    workspace: LightWorkspaceType,
+    transaction?: Transaction
+  ) {
+    return this.model.destroy({
+      where: {
+        workspaceId: workspace.id,
+      },
+      transaction,
+    });
+  }
+
+  async delete(auth: Authenticator): Promise<Result<undefined, Error>> {
+    try {
+      if (this.isReady) {
+        await getPrivateUploadBucket()
+          .file(this.getCloudStoragePath(auth, "original"))
+          .delete();
+
+        // Delete the processed file if it exists.
+        await getPrivateUploadBucket()
+          .file(this.getCloudStoragePath(auth, "processed"))
+          .delete({ ignoreNotFound: true });
+      }
+
+      await this.model.destroy({
+        where: {
+          id: this.id,
+        },
+      });
+
+      return new Ok(undefined);
+    } catch (error) {
+      return new Err(error as Error);
+    }
+  }
+
+  // Status logic.
+
+  private async update(
+    blob: Partial<CreationAttributes<FileModel>>
+  ): Promise<void> {
+    const [, affectedRows] = await this.model.update(blob, {
+      where: {
+        id: this.id,
+      },
+      returning: true,
+    });
+
+    // Update the resource to reflect the new status.
+    Object.assign(this, affectedRows[0].get());
+  }
+
+  async markAsFailed(): Promise<void> {
+    return this.update({ status: "failed" });
+  }
+
+  async markAsReady(): Promise<void> {
+    return this.update({ status: "ready" });
+  }
+
+  get isReady(): boolean {
+    return this.status === "ready";
+  }
+
+  get isCreated(): boolean {
+    return this.status === "created";
+  }
+
+  get isFailed(): boolean {
+    return this.status === "failed";
+  }
+
+  // Cloud storage logic.
+
+  getPublicUrl(auth: Authenticator): string {
+    // TODO(2024-07-01 flav) Remove once we introduce AuthenticatorWithWorkspace.
```

Comment:
this todo really is not needed :)

Reply:
I'll keep around since this check really bothers me and clutters a lot the code.